specification/sql/resource-manager/Microsoft.Sql/preview/2024-02-01-preview/DatabaseBilling.json

```diff
@@ -0,0 +1,124 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2024-02-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/billing/servers/{serverName}/databases/{databaseName}": {
```

Comment:
>billing/servers/{serverName [](http://example.com/codeflow?start=96&length=27)

This does not look like a valid resource path! ARM requires specific path format, see RPC guidelines.

Reply:
Could you clarify intended relationship between servers, databases, and billing? Is 'billing' supposed to be a singleton child of one of the other resource types? (My first guess without any background info)

Reply:
Thank you for the feedback Tim. The relationship is that one server has many databases, and one database has one billingVersion. I am a little confused about how to correctly model this as an azure resource. 

From the guidelines here is a "good" example:

"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MyNameSpace/MyResourceType/{ResourceName}"

And here is my current design which is failing validation:

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/billing/servers/{serverName}/databases/{databaseName}

To comply with the guidelines, do I need to change the design to create a billing "resource" and then have version be a piece of that? Something like a BillingPolicy? Is this the correct format?

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/billing/servers/{serverName}/databases/{databaseName}/BillingPolicy/{BillingPolicyAttribute}